conans/test/unittests/client/tools/cppstd_required_test.py

```diff
@@ -0,0 +1,183 @@
+import unittest
+from mock import mock
+from parameterized import parameterized
+
+from conans.test.utils.conanfile import MockConanfile, MockSettings
+from conans.client.tools import OSInfo
+from conans.errors import ConanInvalidConfiguration
+
+from conans.tools import check_min_cppstd, valid_min_cppstd
+
+
+class UserProneTests(unittest.TestCase):
+
+    def setUp(self):
+        self.conanfile = MockConanfile(MockSettings({}))
+
+    @parameterized.expand(["check_min_cppstd", "valid_min_cppstd"])
+    def test_check_none_cppstd(self, method):
+        """ Cppstd must use a valid number as described in settings.yml
+        """
+        with self.assertRaises(AssertionError) as asserts:
+            eval(method)(self.conanfile, None, False)
+        self.assertEqual("Cannot check invalid cppstd version", str(asserts.exception))
+
+    @parameterized.expand(["check_min_cppstd", "valid_min_cppstd"])
+    def test_check_none_conanfile(self, method):
+        """ conanfile must be a ConanFile object
+        """
+        with self.assertRaises(AssertionError) as raises:
+            eval(method)(None, "17", False)
+        self.assertEqual("conanfile must be a ConanFile object", str(raises.exception))
+
```

Comment:
The part where Conan validates if the cppstd passed by argument or by profile is valid according the list in settings.yml, was executed before by conan create or a related command.

The check_min_cppstd will validate cppstd with the conanfile.settings.compiler.cppstd, which was evaluated early.

Now, for this specific case, if the current settings does not offer cppstd, or compiler.cppstd is higher, it will check if the current compiler version is able to use this version by cppstd_flag. It will fail: "Current compiler does not support the required C++"

Reply:
This kind of situation we can simulate on functional test, like passing `conan create . -s compiler.cppstd=13`